### PR TITLE
support reloadable EnvoyGateway configuration

### PR DESCRIPTION
**Pull request by zirain**
fixes: https://github.com/envoyproxy/gateway/issues/3209

### CODE REVIEW

**Comment by zirain** [comment]
```
k get po -n envoy-gateway-system
NAME                               READY   STATUS    RESTARTS      AGE
envoy-gateway-567bffcffb-znc82     1/1     Running   1 (41m ago)   43m
envoy-ratelimit-7db7dd7c59-gmhkl   1/1     Running   0             41m
```

need to tune the configuration, make sure readiness probe didn't kill the pod before leaser acquired.

File: tools/hack/deployment-exists.sh [hunk 1]
```diff
@@ -3,9 +3,8 @@
 DEPLOYMENT_LABEL_SELECTOR=$1
 DEPLOYMENT_NAMESPACE=$2
 
-
 # Timeout for deployment to exist (in seconds)
-exist_timeout=25
+exist_timeout=300
```
Comment (zirain) [inline tools/hack/deployment-exists.sh:7]: configmap updates to the pod take at least 30 seconds to effect in pod, increase timeout for this.

File: tools/make/kube.mk [hunk 2]
```diff
@@ -138,9 +138,6 @@ e2e: create-cluster kube-install-image kube-deploy install-ratelimit install-e2e
 install-ratelimit:
 	@$(LOG_TARGET)
 	kubectl apply -f examples/redis/redis.yaml
-	kubectl rollout restart deployment envoy-gateway -n envoy-gateway-system
-	kubectl rollout status --watch --timeout=5m -n envoy-gateway-system deployment/envoy-gateway
-	kubectl wait --timeout=5m -n envoy-gateway-system deployment/envoy-gateway --for=condition=Available
```
Comment (zirain) [inline tools/make/kube.mk:143]: EG controller support reload configuration, rollout is unnecessary.

**Review by arkodg — APPROVED** [review]
LGTM thanks !
would be great if you can also attach some manual testing logs to the PR

**Comment by zirain** [comment]
> LGTM thanks ! would be great if you can also attach some manual testing logs to the PR

there's no more manual test, our e2e change the `envoy-gateway-config` during set up.
wasm http server changes catched by another e2e test, which spend me a lot of time to find out.

**Comment by arkodg** [comment]
> > LGTM thanks ! would be great if you can also attach some manual testing logs to the PR
> 
> there's no more manual test, our e2e change the `envoy-gateway-config` during set up. wasm http server changes catched by another e2e test, which spend me a lot of time to find out.

Ah nice ! 

Let's remember to delete all the rollout commands in the docs in a follow PR which we suggest when editing EnvoyGateway config

**Comment by zirain** [comment]
> > > LGTM thanks ! would be great if you can also attach some manual testing logs to the PR
> > 
> > 
> > there's no more manual test, our e2e change the `envoy-gateway-config` during set up. wasm http server changes catched by another e2e test, which spend me a lot of time to find out.
> 
> Ah nice !
> 
> Let's remember to delete all the rollout commands in the docs in a follow PR which we suggest when editing EnvoyGateway config

will update doc after this merged.

**Review by zhaohuabing — APPROVED** [review]
LGTM Thanks!

**Pull request merged**